timer/src/main/java/com/toggl/timer/startedit/ui/StartEditDialogFragment.kt

```diff
@@ -391,27 +391,54 @@ class StartEditDialogFragment : BottomSheetDialogFragment() {
         editDialog = null
     }
 
-    private fun startEditingTime(initialTime: OffsetDateTime) = with(
-        TimePickerDialog(requireContext(), null, initialTime.hour, initialTime.minute, true)
-    ) {
-        setOnCancelListener(dispatchingCancelListener)
-        show()
-        editDialog = this
+    private fun startEditingTime(initialTime: OffsetDateTime) {
+        val onTimeSetListener = TimePickerDialog.OnTimeSetListener { _, hourOfDay, minute ->
+            val changedTime = initialTime
+                .withHour(hourOfDay)
+                .withMinute(minute)
+            store.dispatch(StartEditAction.DateTimePicked(changedTime))
+        }
+        val dialog = TimePickerDialog(
+            requireContext(),
+            onTimeSetListener,
+            initialTime.hour,
+            initialTime.minute,
+            true
+        )
+        with(dialog) {
+            setOnCancelListener(dispatchingCancelListener)
+            show()
+            editDialog = this
+        }
```

Comment:
What do you think about: 
```suggestion
        TimePickerDialog(
            requireContext(),
            onTimeSetListener,
            initialTime.hour,
            initialTime.minute,
            true
        ).run {
            setOnCancelListener(dispatchingCancelListener)
            show()
            editDialog = this
        }
```

It's just a small suggestion to avoid creating that val, but it doesn't make much of a difference.
If you accept this suggestion, do the same for the date picker.